### ISSUE TITLE
Upgrade gradle from v3 -> v8

### ISSUE THREAD

**Issue by mgk** [mgk/s3-plugin#18]
The project is using a very old gradle setup. In order to be published again it needs to use a more recent gradle config.

**Comment by kschults** [comment]
@mgk I opened https://github.com/mgk/s3-plugin/pull/19/files. I wasn't sure how your testing setup works, and I haven't ever worked with gradle plugin development, so I wasn't able to test it out. If you've got some pointers, I can try to dig in again, but hopefully this is enough to get you going in the right direction.

**Comment by luisgoncalves** [comment]
IF the plugin is upgraded to v8, will it become unusable in older versions? I don't know if Gradle prevents its installation. Perhaps if it doesn't rely on APIs that only exist in v8 it can still work if installed in older versions

I'm asking because Gradle 8 is fairly recent.

**Comment by renatoathaydes** [comment]
I don't think you should be very worried because no one needs to upgrade their plugins when they're using very old Gradle versions...

If I remember correctly, there was a big change in the Gradle Plugin SDK between versions 4 and 5, with only minor improvements after that. So you could get away with something like Gradle 5, but I think that it's not worth it because most people who were already on Gradle 5 could, without much trouble, upgrade their projects to newer Gradle versions.

**Comment by luisgoncalves** [comment]
I guess https://github.com/mgk/s3-plugin/pull/19 solved this ticket.

**Comment by luisgoncalves** [comment]
@mgk would it be possible to have a released version/variant targeting Java 11? We still have some older build agents running Gradle with Java 11..

**Comment by kschults** [comment]
@luisgoncalves It looks like this still builds targeting Java 8: https://github.com/mgk/s3-plugin/blob/main/gradle.properties#L4. That should be okay for your old versions, I think.

**Comment by luisgoncalves** [comment]
I actually tried it with Gradle 8.5 running with Java 11, and got:

```
10:38:17  Caused by: org.gradle.internal.component.NoMatchingGraphVariantsException: No matching variant of com.github.mgk.gradle:s3:1.6.0 was found. The consumer was configured to find a library for use during runtime, compatible with Java 11, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.5' but:
10:38:17    - Variant 'apiElements' capability com.github.mgk.gradle:s3:1.6.0 declares a library, packaged as a jar, and its dependencies declared externally:
10:38:17        - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
10:38:17        - Other compatible attribute:
10:38:17            - Doesn't say anything about org.gradle.plugin.api-version (required '8.5')
```

From the [Java plugin docs](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension), it seems that the default values for `sourceCompatibility` and `targetCompatibility` on the `java` extension rely on the `toolchain`, which by default is inferred from the Java version used to execute Gradle.

Perhaps the Gradle properties are not considered anymore? I skimmed through release notes and couldn't find mention of that..

**Comment by kschults** [comment]
Certainly could be the case. Probably needs to get defined in `build.gradle`. I don't think the code is doing anything that would be incompatible, so if you open up a new PR to do it, hopefully MGK would be able to merge and deploy it.

**Comment by luisgoncalves** [comment]
@kschults did as suggested: https://github.com/mgk/s3-plugin/pull/20

Works in a local smoke test, so the theory of `targetCompatibility` not being picked up from the properties file seems to hold.

@mgk Do you think you can check the PR, and release a fix that is actually built targeting Java 8? Thanks in advance.
